### PR TITLE
Move should_log log_bitmask checking into Dataflash

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: AntennaTracker/system.cpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ void Tracker::init_tracker()
     }
 
     // GPS Initialization
-    gps.init(nullptr, serial_manager);
+    gps.init(serial_manager);
```
Comment (OXINARF) [inline AntennaTracker/system.cpp]: Tracker has MASK_LOG_GPS, any reason not to set it? I know it is a change in behaviour, but I can't see a reason why Tracker wouldn't want to behave like other vehicles.

File: Tools/Replay/Replay.h [hunk 2]
```diff
@@ -69,7 +69,7 @@ class ReplayVehicle {
     AP_InertialNav_NavEKF inertial_nav{ahrs};
     AP_Vehicle::FixedWing aparm;
     AP_Airspeed airspeed;
-    DataFlash_Class dataflash{"Replay v0.1"};
+    DataFlash_Class dataflash{"Replay v0.1", 0}; // logging is magic for Replay
```
Comment (OXINARF) [inline Tools/Replay/Replay.h]: How does this work? You should be passing an AP_Int32 reference. And wouldn't 0 prevent any logging at all?

File: libraries/AP_InertialSensor/AP_InertialSensor.h [hunk 3]
```diff
@@ -202,10 +202,7 @@ class AP_InertialSensor : AP_AccelCal_Client
     uint8_t get_accel_filter_hz(void) const { return _accel_filter_cutoff; }
 
     // pass in a pointer to DataFlash for raw data logging
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor.h]: Will fix.

File: libraries/AP_InertialSensor/AP_InertialSensor.h [hunk 4]
```diff
@@ -202,10 +202,7 @@ class AP_InertialSensor : AP_AccelCal_Client
...
-    void set_dataflash(DataFlash_Class *dataflash) { _dataflash = dataflash; }
-
-    // enable/disable raw gyro/accel logging
-    void set_raw_logging(bool enable) { _log_raw_data = enable; }
+    void set_log_raw_log_bitmask_bit(uint32_t log_raw_log_bitmask_bit) { _log_raw_log_bitmask_bit = log_raw_log_bitmask_bit; }
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor.h]: Will change as you suggest.  Yours didn't turn my brain inside out, so I'll rename.

File: libraries/AP_GPS/AP_GPS.h [hunk 5]
```diff
@@ -370,10 +369,9 @@ class AP_GPS
 
     void Write_DataFlash_Log_Startup_messages();
 
-protected:
+    void set_log_gps_log_bitmask_bit(uint32_t mask) { _log_gps_log_bitmask_bit = mask; }
```
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS.h]: Yep, changing to `bit`

File: AntennaTracker/system.cpp (hunk 1 above)
Comment (peterbarker) [inline AntennaTracker/system.cpp]: Will add a commit to change behaviour.

File: Tools/Replay/Replay.h (hunk 2 above)
Comment (peterbarker) [inline Tools/Replay/Replay.h]: It doesn't work .  My testing was faulty.

Fixed.

**Comment by OXINARF** [comment]
Rebased, conflicts solved, merged! Thanks!

**Pull request closed without merging**